lib/vrl/stdlib/src/encode_logfmt.rs

```diff
@@ -0,0 +1,227 @@
+use vrl::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct EncodeLogfmt;
+
+impl Function for EncodeLogfmt {
+    fn identifier(&self) -> &'static str {
+        "encode_logfmt"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "value",
+            kind: kind::OBJECT | kind::ARRAY,
+            required: true,
+        }]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Compiled {
+        let value = arguments.required("value");
+
+        Ok(Box::new(EncodeLogfmtFn { value }))
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[
+            Example {
+                title: "encode object",
+                source: r#"encode_logfmt({"lvl": "info", "msg": "This is a message", "log_id": 12345})"#,
+                result: Ok(r#"s'log_id=12345 lvl=info msg="This is a message"'"#),
+            },
+            Example {
+                title: "encode array",
+                source: r#"encode_logfmt([["lvl", "info"], ["msg", "This is a message"], ["log_id", 12345]])"#,
+                result: Ok(r#"s'lvl=info msg="This is a message" log_id=12345'"#),
+            },
+        ]
+    }
+}
+
+fn format_logfmt_string(f: &mut fmt::Formatter<'_>, str: &str) -> fmt::Result {
+    match str.find(' ') {
+        Some(_) => write!(f, r#""{}""#, str),
```

Comment:
Do we need to encode any `"`s appearing in the value?

Reply:
Good catch, I think we should escape strings the same way json strings are escaped